resharper/resharper-unity/src/HlslSupport/UnityGameEngineDirectiveResolver.cs

```diff
@@ -0,0 +1,115 @@
+        }
+        
+        public bool IsApplicable(CppInclusionContext context, string path)
+        {
+              return path.StartsWith("Packages/") &&
+                   (mySolutionTracker.IsUnityProject.HasTrueValue() || myReferencesTracker.HasUnityReference.HasTrueValue());
+        }
+
+        public string TransformPath(CppInclusionContext context, string path)
+        {
+            var pos = path.IndexOf('/') + 1;
+            if (pos == -1)
+                return path;
+            var endPos = path.IndexOf('/', pos);
+            if (endPos == -1)
+                endPos = path.Length;
+
+            var packageName = path.Substring(pos, endPos - pos);
+
+            var version = GetVersionFor(packageName);
+            var localPackagePath = mySolution.SolutionDirectory.Combine("Packages").Combine(packageName + "@" + version + path.Substring(endPos));
+            if (localPackagePath.Exists == FileSystemPath.Existence.File)
+                return localPackagePath.FullPath;
+            
+            var cachedPackagePath = mySolution.SolutionDirectory.Combine("Library").Combine("PackageCache").Combine(packageName + "@" + version + path.Substring(endPos));
+            return cachedPackagePath.FullPath;
+        }
+
+        private readonly object myLockObject = new object();
+        // TODO, add package version tracker
+        private ConcurrentDictionary<string, string> myPackageVersions = null;
+        
+        [NotNull]
+        private string GetVersionFor(string packageName)
+        {
+            if (myPackageVersions == null)
+            {
+                lock (myLockObject)
+                {
+                    if (myPackageVersions == null)
+                    {
+                        var result = new ConcurrentDictionary<string, string>();
+                        try
+                        {
+                            var solDir = mySolution.SolutionDirectory;
+                            var packagesFolder = solDir.Combine("Packages");
+                            var packagesCacheFolder = solDir.Combine("Library").Combine("PackageCache");
+
+                            var candidates = new[] {packagesFolder, packagesCacheFolder};
+                            foreach (var candidate in candidates)
+                            {
+                                foreach (var folder in candidate.GetDirectoryEntries())
+                                {
+                                    if (!folder.IsDirectory)
+                                        continue;
+                                        
+                                    var packageAndVersion = folder.RelativePath.Name.Split('@');
+                                    if (packageAndVersion.Length != 2)
+                                        continue;
+                                    
+                                    if (result.ContainsKey(packageAndVersion[0]))
+                                        continue;
```

Comment:
You could use `JetSemanticVersion` on the second element and always use the highest version.